src/common/libkvs/kvs_txn.c

```diff
@@ -176,7 +176,7 @@ int flux_kvs_txn_put (flux_kvs_txn_t *txn, int flags,
     }
     if (validate_flags (flags, 0) < 0)
         goto error;
-    if (!(dirent = treeobj_create_val (value, value ? strlen (value) + 1 : 0)))
+    if (!(dirent = treeobj_create_val (value, value ? strlen (value) : 0)))
         goto error;
     if (append_op_to_txn (txn, flags, key, dirent) < 0)
         goto error;
```

Comment:
Yes but I didn't do that because I think it reads a bit easier if each function is variation with a common structure.  I had them calling each other before to avoid code duplication and I thought it maybe was a step in the wrong direction.  I don't feel too strongly though, so if you do, I don't mind changing it, LMK.

Reply:
Nah, we can leave as is then.